src/main/java/com/microsoft/jfr/FlightRecorderConnection.java

```diff
@@ -113,19 +100,12 @@ public long startRecording(RecordingOptions recordingOptions, RecordingConfigura
             String[] argTypes = new String[]{};
             final long id = (long) mBeanServerConnection.invoke(objectName, "newRecording", args, argTypes);
 
-            if (recordingConfiguration != null) {
-                String configuration = recordingConfiguration.getConfiguration();
-                if (configuration != null && configuration.trim().length() > 0) {
-                    args = new Object[]{id, configuration};
-                    argTypes = new String[]{long.class.getName(), String.class.getName()};
-                    mBeanServerConnection.invoke(objectName, recordingConfiguration.getMbeanSetterFunction(), args, argTypes);
-                }
-            }
+            setOptions(recordingConfiguration, id);
```

Comment:
I think `if (recordingConfiguration != null) { setConfiguration(recordingConfiguration, id); }` is better because it documents the behavior of the startRecording method.

Reply:
Done